go.mod

```diff
@@ -10,9 +10,18 @@ require (
 	github.com/golang/protobuf v1.5.3
 	github.com/grpc-ecosystem/grpc-gateway v1.16.0
 	github.com/stretchr/testify v1.8.4
+	google.golang.org/genproto/googleapis/api v0.0.0-20231009173412-8bfb1ae86b6c
+	google.golang.org/grpc v1.58.3
```

Comment:
Was confused at first by title of PR, but this is really "Bump Go to 1.20 and upgrade gRPC library"

Can you give more details on the need to upgrade gRPC? (not that I disagree, just want to understand)

Reply:
I can revert the library updates if you want and only bump Go, but the thing is since `go mod tidy` is part of the Makefile target to generate the files, the next person will inevitably update this. That's actually what led me here to having to bump the Go version.

Reply:
I wonder if we want to decouple "generate files" from "bump versions"? It's a little odd that they are so tightly coupled.

This is the part of the Makefile that I'm referring to:
```
# Compile proto files.
proto: grpc goimports proxy grpc-mock copyright

# Update submodule and compile proto files.
update-proto: update-proto-submodule proto update-dependencies gomodtidy
```
Notice how `update-dependencies` is part of `update-proto`.

Reply:
Nevermind, I understand how the workflow of this and https://github.com/temporalio/api work now; I've reverted any library updates and made this PR focus on the Go version bump!